### ISSUE TITLE
hat sich die mediathek api geändert?

### ISSUE THREAD

**Issue by gamedog47** [dicer/auto-tatort#19]
Hi, 
seit ca. einer Woche funktioniert die URL aus der Beispiel-Config nicht mehr.

Status-Code: 410 "Gone"

Welche API Urls muss ich verwenden, damit das Skript wieder läuft?

Vielen Dank für das coole Skript und Unterstützung.

**Comment by dicer** [comment]
Leider hat die ARD die Mediathek auf eine neue Oberflaeche umgestellt.
Dort scheint es keine RSS Feeds mehr zu geben, auf die ich bisher
zurueck gegriffen habe.
Ich schaetze auch, dass das so bleibt. Ich habe daher vor das Script auf
die Feeds von mediathekviewweb.de umzustellen. Da ist es aber ein ganzes
Stueck schwerer nur die Tatortfolgen zu erwischen (die Suche nach Tatort
foerdert da auch eine Menge Sachen zu Tage die man nicht laden will).

Ich weiss von dem Problem und bekomme auch alle 2h eine Erinnerungsmail
von meinem Script. Ich werde mich bald drum kuemmern. Kann aber noch
keinen Termin versprechen.

**Comment by doczkal** [comment]
@dicer konntest du dir das Script in der Vergangenheit nochmal anschauen?

Ich hab auf der MediathekViewWeb Seite ein wenig mit den Einstellungen gespielt und bin mit "!ARD !Tatort +Tatort: #Tatort >80" in der Suche zu einem passablen Ergebnis gekommen.

**Comment by Spanishu** [comment]
Hallo,
ich hoffe es geht Euch allen gut. Gibt es eine chance das wir das tool gefixed bekommen können?

**Comment by gamedog47** [comment]
Hi, ich habe das Skript vor ein paar Woche ( wie von @dicer empfohlen) auf mediathekviewweb.de umgebaut. Für mich funktioniert es gut, aber leider ...
- sieht mein Code jetzt nicht mehr so schön aus
- ich musste einige Funktionen abschalten
- (und er läuft auch nicht mehr unter Python 2 ist für mich momentan noch wegen Kodi erforderlich)

Ich glaube für Dich (@dicer) ist es wohl nur eine Fingerübung.

Viele Grüße

**Comment by dicer** [comment]
Ich will das ganze eigentlich komplett neu schreiben, da bei
mediathekviewweb (MVW) im RSS Feed direkt die fertigen Downloadlinks
drin sind. Das erübrigt eigentlich die meiste Logik im Script, welche
sich um das Discovery der Videos auf den ARD Seiten gekümmert hat.

Andererseits ist es um einiges schwieriger bei MVW zu filtern, um zum
Beispiel nur den aktuellen Tatort Sonntag Abends zu bekommen.

Daher habe ich eher einen Rewrite geplant. Aber der scheitert gerade an
der Zeit.

Zusätzlich möchte ich auch meine Toolchain integrieren, welche ich
aktuell benutze um die Downloads in Kodi rein zu bekommen:

- Aufbau einer Datenbank aller aktuellen "Folgen" (bei Wikipedia ist das
sehr aktuell)
- Automatisches Umbenennen mit Tatortnummer im Namen
- Generierung der notwendigen Filestruktur (Aufteilung in "Seasons")
damit Kodi das automatisch eingelesen bekommt

Die drei Punkte benutze ich schon eine ganze Weile, aber das ist in
einem "wilden" Zustand und teilweise Bash, teilweise Python usw...

Daher der Plan eines kompletten Rewrites. Das wird aber noch eine Weile
dauern. Ich glaube nur einen Rewrite um einen bestimmten RSS-Feed zu
laden macht nicht so viel Sinn. Da gibt es auch diverse andere RSS
Downloader da draußen die man scripten kann.

**Comment by dicer** [comment]
Hab gerade die URL in der Beispielconfig angepasst. Dank @glitsj16 weiß ich nun, dass aktuell wohl die folgende URL funktioniert:
`https://classic.ardmediathek.de/tv/Tatort/Sendung?documentId=602916&bcastId=602916&rss=true` (http sollte auch gehen)
Keine Ahnung wie lange das noch geht, aber bis dahin kann man das Script erstmal weiter betreiben.

Der oben genannte Rewrite ist noch nicht angefangen... Keine Zeit und ich bin dazu übergegangen die Folgen mittels mediathekviewweb von Hand runter zu laden.

(Alternativlink: https://classic.ardmediathek.de/export/rss/id=602916)

**Comment by thomasmerz** [comment]
Bei mir funzt leider gar keine der o.g. Lösungen… Hat sich schon jemand damit weiter befasst, um das Tool wieder funktionsfähig zu bekommen?
```
🦎🖥  ✔ ~/dev/github/auto-tatort [master|✔]
18:06 $ python3 autoTatort.py
Could not connect to link 'https://classic.ardmediathek.de/tv/Tatort/Sendung?documentId=602916&bcastId=602916&rss=true'.
<unknown>:6:183: not well-formed (invalid token)
🦎🖥  ✔ ~/dev/github/auto-tatort [master|✔]
18:06 $ vim config.json
🦎🖥  ✔ ~/dev/github/auto-tatort [master|✔]
18:06 $ python3 autoTatort.py
Could not connect to link 'https://classic.ardmediathek.de/export/rss/id=602916'.
<unknown>:6:183: not well-formed (invalid token)
🦎🖥  ✔ ~/dev/github/auto-tatort [master|✔]
18:06 $
```

**Comment by dicer** [comment]
Am besten nimmste einen RSS Reader der Mediadownloads (`enclosure`) unterstützt und gehst dann gegen mediathekviewweb.de
Dort gibts RSS Feeds und du hast direkt den Link zum mp4. Eine speziellen Downloader sollte es eigentlich nicht mehr brauchen dadurch.
Sowas hier sollte funktionieren: https://github.com/jrabbit/hitman
(habs aber nicht getestet)
Siehe auch: https://github.com/topics/rss-downloader

**Comment by thomasmerz** [comment]
Och, so ein automatischer Download einmal die Woche ist schon was feines und allemal besser als Klicki-klicki 😄 Aber notfalls geht es ja in der Tat auch ohne so einen Automaten.

**Comment by dicer** [comment]
Ja, automatischer Download ist durchaus praktisch. Mit "speziellen Downloader" meinte ich auto-tatort. Den braucht man mittlerweile nicht mehr, weil man Media-Links nicht mehr zusammen puzzeln bzw. aus irgendwelchen Dateien rausholen muss. 

Man kann einfach einen RSS-Downloader nehmen kann. Den kannste ja auch einmal die Woche anwerfen und dann zieht er halt runter was der MWV-RSS-Feed so hergibt. Liegt dann nur an dir den Feed vernünftig zu konfigurieren. Das ist gar nicht mehr so einfach, weil es nicht klar ist, was Wiederholungen und was neue Filme sind.
